driver.go

```diff
@@ -738,6 +753,17 @@ func (d *Driver) WaitTask(ctx context.Context, taskID string) (<-chan *drivers.E
 		return nil, drivers.ErrTaskNotFound
 	}
 	ch := make(chan *drivers.ExitResult)
+	// only start logstreamer if we have to...
+	var driverConfig TaskConfig
+	if err := handle.taskConfig.DecodeDriverConfig(&driverConfig); err != nil {
+		d.logger.Warn("Unable to decode driver config, not starting log streamer", "task", taskID, "error", err)
+	} else {
+		// start to stream logs if journald log driver is configured and LogCollection is not disabled
+		if driverConfig.Logging.Driver == LOG_DRIVER_JOURNALD && !d.config.DisableLogCollection {
+			go handle.runLogStreamer(ctx)
+		}
+	}
+
```

Comment:
This branch is somewhat old and i forgot about that detail. But i can imagine that it was a timing related issue. Let me run a few tests. It would be cleaner to start the logger in StartTask().

Reply:
It is there because i can rely on nomad that both a new and a recovered task will run through this method. I could move it out but this would require me to duplicate some code. During my tests i found that the task recovery was broken, fixed it now. So code is slightly different.
Can we keep it in WaitTask?

Reply:
Ahh got it. Yeah, I think it's fine to leave it in `WaitTask` 👍 

> During my tests i found that the task recovery was broken, fixed it now.

Was this a problem in plugin or in Nomad?